### DISCUSSION TITLE
`options(languageserver.xxxx = yyyyy)` rename to `options(r.lsp.xxxx = yyyyy)`

### DISCUSSION THREAD

**Discussion by randy3k** [REditorSupport/languageserver#441] [General]
just try to get some feedback to see what setting names are better.

Currently, LSP configuration are set via the dotted variable `r.lsp.xxxx` which essentially means in JSON
```
"r": {
  "lsp": {
        settings here
  }
}
```

On the other hand, the `options` method has names `languageserver.xxxxx`.
Perhaps it would be more consistent if we change the names to `r.lsp.xxxxx`?! One potential drawback that I would think of is that users may be confused between the difference in setting values of JSON and R.

@renkun-ken what you think?

**Comment by renkun-ken** [comment]
I thought we didn't use `r.lsp.*` in the names of the R options because they are only specific to vscode-r-lsp client?

As for the confusion between the different representation of values between JSON and R, I think it will be confusing. Another potential confusion is that if we add these settings to vscode-r-lsp and we should probably use the camelCase convention like all other vscode settings, then the settings will be `r.lsp.richDocumentation`, `r.lsp.snippetSupport`, `r.lsp.maxCompletions`, `r.lsp.lintCache`, etc..

**Reply by randy3k** [reply]
Good point. Perhaps we should deprecate the snake case and switch to camelCase
